### PR TITLE
Module 1 project exercise

### PR DESCRIPTION
**Pull request by pcoulso1**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -6,7 +6,31 @@
...
+@app.route('/add', methods=['GET','POST'])
+def add():
+    if request.method == 'POST':
+        if 'add' in request.form:
+            session.add_item(request.form.get('new_todo'))
+        return redirect("/")
+
+    return render_template('add.html', items=session.get_items())
+
+@app.route('/delete', methods=['POST'], defaults={'id': None})
```
Comment (Jonesey13) [inline app.py:25]: Nice use of multiple routes!

File: app.py [hunk 2]
```diff
@@ -6,7 +6,31 @@
...
-    return 'Hello World!'
+    return render_template('index.html', items=session.get_items())
 
+@app.route('/update/<id>')
+def update(id):
+    session.update_item(id)
+    return redirect("/")
+
+@app.route('/add', methods=['GET','POST'])
+def add():
```
Comment (Jonesey13) [inline app.py:17]: You could probably add some server side validation here (to catch empty todos)

File: session_items.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 from flask import session
+from collections import OrderedDict
```
Comment (Jonesey13) [inline session_items.py]: I don't think you need this line?

File: session_items.py [hunk 4]
```diff
@@ -67,3 +70,34 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def update_item(id):
```
Comment (Jonesey13) [inline session_items.py]: Maybe a better name for this method is `complete_item`?

File: templates/delete.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{% extends "layout.html" %}
+{% block title %}To-Do App{% endblock %}
+
+{% block navbar %}
+<div class="collapse navbar-collapse" id="navbarText">
```
Comment (Jonesey13) [inline templates/delete.html]: When using `{% Jinja expression %}` I'd indent its content (i.e the `div` below) one level in

File: templates/add.html [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+
+{% block content %}
+  <div class="row">
+    <div class="col-md-12">
+      <form action="/add" method="post">
+        <h2>Add New Items</h2>
+        <div class="form-group">
+          <label for="todo">Description</label>
+          <input type="text" class="form-control" id="todo" name="new_todo">
```
Comment (Jonesey13) [inline templates/add.html]: Probably worth adding a `required` attribute here for some server side validation.

**Review by Jonesey13 — APPROVED** [review]
Looks all functionally sound to me!

Made a few comments about styling but feel free to merge this PR in whenever you are ready :)

File: session_items.py (hunk 4 above)
Comment (pcoulso1) [reply session_items.py]: Renamed

File: templates/delete.html (hunk 5 above)
Comment (pcoulso1) [reply templates/delete.html]: Done

**Comment by pcoulso1** [comment]
Merging pull request

**Pull request merged**